### PR TITLE
Add PyTest test for sigmoid binary cross entropy layer

### PR DESCRIPTION
**Pull request by timmoon10**
I've also fixed some mistakes from copy-pasting.

[Bamboo is green](https://lc.llnl.gov/bamboo/browse/LBANN-TIM248-1).

### CODE REVIEW

File: bamboo/unit_tests/test_unit_layer_sigmoid_binary_cross_entropy.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    obj.append(z)
+    metrics.append(lbann.Metric(z, name='data-parallel layout'))
+
+    # NumPy implementation
+    vals = []
+    for i in range(num_samples()):
+        x = get_sample(i).astype(np.float64)
+        x0 = x[:slice_size]
+        x1 = x[slice_size:]
+        y = -x1 * np.log1p(np.exp(-x0)) - (1-x1) * np.log1p(np.exp(x0))
```
Comment (samadejacobs) [inline bamboo/unit_tests/test_unit_layer_sigmoid_binary_cross_entropy.py]: This doesnt exactly match x1 * log(1 + exp(-x0)) + (1 - x1) * (x0 + log(1 + exp(-x0)) from [TF](https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits?version=stable)? Just checking, I could be wrong.

Comment (timmoon10) [reply bamboo/unit_tests/test_unit_layer_sigmoid_binary_cross_entropy.py]: It's mathematically the same. They rearranged stuff to avoid numerical errors, but this test isn't in a regime where that is a big concern.
